packages/next/build/swc/src/next_dynamic.rs

```diff
@@ -0,0 +1,174 @@
+use swc_atoms::{js_word, JsWord};
+use swc_common::{FileName, DUMMY_SP};
+use swc_ecmascript::ast::{
+  ArrayLit, ArrowExpr, BinExpr, BinaryOp, BlockStmtOrExpr, CallExpr, Expr, ExprOrSpread,
+  ExprOrSuper, Ident, ImportDecl, ImportSpecifier, KeyValueProp, Lit, MemberExpr, ObjectLit, Prop,
+  PropName, PropOrSpread, Str, StrKind,
+};
+use swc_ecmascript::visit::Fold;
+
+pub fn next_dynamic(filename: FileName) -> impl Fold {
+  NextDynamicPatcher {
+    filename,
+    dynamic_binding: js_word!(""),
+  }
+}
+
+#[derive(Debug)]
+struct NextDynamicPatcher {
+  filename: FileName,
+  dynamic_binding: JsWord,
+}
+
+impl Fold for NextDynamicPatcher {
+  fn fold_import_decl(&mut self, decl: ImportDecl) -> ImportDecl {
+    let ImportDecl {
+      ref src,
+      ref specifiers,
+      ..
+    } = decl;
+    if &src.value == "next/dynamic" {
+      for specifier in specifiers {
+        if let ImportSpecifier::Default(default_specifier) = specifier {
+          self.dynamic_binding = default_specifier.local.sym.clone();
+        }
+      }
+    }
+
+    decl
+  }
+
+  fn fold_call_expr(&mut self, mut expr: CallExpr) -> CallExpr {
+    if let ExprOrSuper::Expr(i) = &expr.callee {
+      if let Expr::Ident(Ident { sym, .. }) = &**i {
+        if sym == &self.dynamic_binding {
+          let mut import_specifier = None;
+          if let Expr::Arrow(ArrowExpr {
+            body: BlockStmtOrExpr::Expr(e),
+            ..
+          }) = &*expr.args[0].expr
```

Comment:
This will panic if the user does not pass any argument to this call.

There are several ways for handling this error.

react jsx pass of swc reports error via `swc_emca_utils::HANDLER`, like https://github.com/swc-project/swc/blob/48bc26d3c93e64174b3d1cd269205a98b1581a0c/ecmascript/transforms/react/src/jsx/mod.rs#L411-L423
This will print a nice-looikng error on stderr, as it's configured while creating `swc_common::Handler`. (See https://github.com/vercel/next.js/blob/d368f76b4dff84a2db46ee50a09ae29003f11b30/packages/next/build/swc/src/lib.rs#L56-L61)

You can also store the error in the struct (`NextDynamicPatcher `), and check for it after applying transforms. You can panic or pass error to napi task defined at https://github.com/vercel/next.js/blob/d368f76b4dff84a2db46ee50a09ae29003f11b30/packages/next/build/swc/src/transform.rs#L66-L79
This seems unpractical to me.

Alternatively, you can simply panic and it will be an exception for javascript.